transport/ice/IceSession.cpp

```diff
@@ -873,13 +873,17 @@ bool IceSession::isIceComplete(const uint64_t now)
     {
         return false;
     }
-    if (now - _sessionStart > _config.connectTimeout * utils::Time::ms)
+    if (utils::Time::diffGT(_sessionStart, now, _config.connectTimeout * utils::Time::ms))
     {
         return true;
     }
 
-    if (failCount > 0 && failCount == _checklist.size())
+    // if all candidates failed, it could be that we just had unreachable HOST candidates. Give it some time for a PRFLX
+    // to show up.
+    if (failCount > 0 && failCount == _checklist.size() &&
+        probeAge > static_cast<int64_t>(_config.additionalCandidateTimeout * utils::Time::ms))
```

Comment:
Without time check here thw prflx must arrive before the first probe (that is host and will fail) has failed.
Usually that does occure but if it is tcp or something it can fail fast.